src/voice-broadcast/hooks/useHasRoomLiveVoiceBroadcast.ts

```diff
@@ -14,18 +14,26 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { useState } from "react";
+import { useContext, useState } from "react";
 import { Room, RoomStateEvent } from "matrix-js-sdk/src/matrix";
 
 import { hasRoomLiveVoiceBroadcast } from "../utils/hasRoomLiveVoiceBroadcast";
 import { useTypedEventEmitter } from "../../hooks/useEventEmitter";
+import { SDKContext } from "../../contexts/SDKContext";
 
 export const useHasRoomLiveVoiceBroadcast = (room: Room) => {
-    const [hasLiveVoiceBroadcast, setHasLiveVoiceBroadcast] = useState(hasRoomLiveVoiceBroadcast(room).hasBroadcast);
-
-    useTypedEventEmitter(room.currentState, RoomStateEvent.Update, () => {
-        setHasLiveVoiceBroadcast(hasRoomLiveVoiceBroadcast(room).hasBroadcast);
-    });
-
+    const sdkContext = useContext(SDKContext);
+    const [hasLiveVoiceBroadcast, setHasLiveVoiceBroadcast] = useState(false);
+
+    const update = sdkContext.client
+        ? () => {
+              hasRoomLiveVoiceBroadcast(sdkContext.client!, room).then(({ hasBroadcast }) => {
+                  setHasLiveVoiceBroadcast(hasBroadcast);
+              });
+          }
+        : () => {}; // noop without client
+
+    update();
```

Comment:
We want to call update at every render of the hook ?

Reply:
Good catch. No, we don't want to.

My issue is that `hasRoomLiveVoiceBroadcast` is now async. Means I cannot use it as initial value for the state any more. What I want to do here is trigger `update` once to „initialise“ the state. I would appreciate any suggestion here.

Reply:
In case of async work, I'm afraid that the solution is to use a `useEffect`.

```
useEffect(() => {
   async function checkLiveVoiceBroadcast() {
        const  {hasBoardcast} = await hasRoomLiveVoiceBroadcast(sdkContext.client!, room)
        setHasLiveVoiceBroadcast(hasBroadcast)
   } 

    if (sdkContext.client) {
           checkLiveVoiceBroadcast() 
    }
}, [sdkContent, setHasLiveVoiceBroadcast]
```

I don't have all the the behaviour in mind but something in that way can do the job ?

Reply:
I think this also needs a `useMemo` additionally. Can you have another look at what I did @florianduros  https://github.com/matrix-org/matrix-react-sdk/pull/9780/files#diff-629d18b74dfc90d298e91431dab3af5c859d6c7cd1cbd0046ec51daa799d73b4R28?

`update` should also be called on emitted events.

Reply:
Yes, I think that's good!

I missed the `update` on emitted events (it create also a function in emitted events, maybe a useCallback is also needed)

Reply:
Doesn't [`useTypedEventEmitter`](https://github.com/matrix-org/matrix-react-sdk/blob/cdd4ab56d7aee0db2facdea6effe068af53c7ceb/src/hooks/useEventEmitter.ts#L24) take care of this?

Reply:
Yes indeed
It's store the handler in the ref and every time the handler props changes, the ref is  updated. It's strange way to do it :)